src/rolling/directory.rs

```diff
@@ -266,7 +300,17 @@ impl BlockWrite for RollingWriter {
     }
 
     async fn flush(&mut self) -> io::Result<()> {
-        self.file.flush().await
+        let current_file_number = self.file_number.file_number();
+        tokio::try_join!(
+            async {
+                self.file.flush().await?;
+                self.file.get_ref().sync_data().await
+            },
+            self.directory
+                .sync_data(self.last_fsync..current_file_number)
+        )?;
```

Comment:
Does it help compared to 

```suggestion
          self.file.flush().await?;
          self.directory.sync_data(self.last_fsync..=current_file_number)?;
```

Reply:
in theory mine can be slightly faster (latency wise), we start fsyncing old files while flush is still running for the current one